plugins/source_testing.go

```diff
@@ -101,53 +93,23 @@ func validateResources(t *testing.T, resources []*schema.Resource) {
 	table := resources[0].Table
 
 	// A set of column-names that have values in at least one of the resources.
-	columnsWithValues := make(map[string]bool)
+	columnsWithValues := make([]bool, len(table.Columns))
 
 	for _, resource := range resources {
-		// we want to marshal and unmarshal to mimic over-the-wire behavior
-		b, err := json.Marshal(resource.Data)
-		if err != nil {
-			t.Fatalf("failed to marshal resource data: %v", err)
-		}
-		var data map[string]interface{}
-		if err := json.Unmarshal(b, &data); err != nil {
-			t.Fatalf("failed to unmarshal resource data: %v", err)
-		}
-
-		for columnName, value := range data {
+		for i, value := range resource.GetValues() {
 			if value == nil {
 				continue
 			}
-
-			columnsWithValues[columnName] = true
-
-			switch resource.Table.Columns.Get(columnName).Type {
-			case schema.TypeJSON:
-				switch value.(type) {
-				case string, []byte:
-					t.Errorf("table: %s JSON column %s is being set with a string or byte slice. Either the unmarhsalled object should be passed in, or the column type should be changed to string", resource.Table.Name, columnName)
-					continue
```

Comment:
This check has caught a number of bugs already, but seems like it's missing from the new version. Does the JSON type disallow strings / bytes being passed to it?

Reply:
our new type system knows to take care of that so it tries to unmarshal and if it doesn't work it panics and if not then then it just stores it and the user doesn't need to take care of that.

Reply:
Nevermind - I checked the JSON type definition now and it looks like it's handled there 👍  (It checks if the type is string or byte and then doesn't call json.Marshal on it)

Reply:
@yevgenypats Nice! 👍